project-base/storefront/helpers/cookies/cookiesStoreUtils.ts

```diff
@@ -0,0 +1,34 @@
+import { getCookies, setCookie } from 'cookies-next';
+import { OptionsType } from 'cookies-next/lib/types';
+import { GetServerSidePropsContext } from 'next';
+import { useEffect } from 'react';
+import { CookiesStoreState, useCookiesStore } from 'store/useCookiesStore';
+
+export const COOKIES_STORE_NAME = 'cookiesStore' as const;
+
+export type CookiesType = {
+    cookiesStore: string;
+};
+
+export const getCookiesStore = (context?: GetServerSidePropsContext) => {
+    const { cookiesStore } = getCookies(context) as CookiesType;
+
+    return cookiesStore ? decodeURIComponent(cookiesStore) : null;
+};
+
+export const useCookiesStoreSync = () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { setCookiesStoreState: setCookieStoreValue, ...storeValues } = useCookiesStore((state) => state);
+
+    useEffect(() => {
+        setCookieStore(storeValues);
+    }, [storeValues]);
+};
+
+const setCookieStore = (store: CookiesStoreState, options?: OptionsType) => {
+    const isStoreWithValues = !!Object.values(store).filter((storeValue) => storeValue !== undefined).length;
+
+    if (isStoreWithValues) {
+        setCookie(COOKIES_STORE_NAME, store, { maxAge: 60 * 60 * 24 * 30, ...options });
```

Comment:
It would be nice to store the number in a constant, such that 

`const thirtyDaysInSeconds`

Reply:
Good idea. Implemented.